elastic/kibana
# theme function

## Summary

Adds a `theme` function as proposed here https://github.com/elastic/kibana/issues/71064

Currently theme information (variables) are part of expression variables under `theme` key. Preferably we introduce a service in the future to provide this information.

How to test:
provide a `theme` variable to expressions and see your fonts change:
- `theme` variable needs to be sent to executor with the following structure:
```
theme: {
  font: {
    family: string,
    size: number,
    color: string,
    italics: boolean,
    underline: boolean
    weight: string;
  }
}
```
- every function using `font` function for any of its arguments where specific `font` function arguments have not been overridden will not start using the theme information provided

looking at example canvas workpads it seems canvas always sets size family color and alignment when adding a font property to element, which should probably be changed, so it only adds {font} and all the arguments are set only after user changes from whats set on theme.

also, all the example workpads have most of the font arguments set (which means that a theme won't apply).

followup:
- [ ] theme service would be preferred over using variables
- [ ] align on palette information, update palette function
- [ ] updates on canvas side on how the variables/defaults are set thru UI

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
